Add documentation to appconfig

maririos opened the pull request.

maririos committed:
add documentation

dsgouda commented:
@maririos  Please pull from latest master and update the PR to fix the Travis failures

pakrym commented on src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationSetting.cs:
```diff
@@ -25,8 +34,8 @@ public ConfigurationSetting(string key, string value, string label = null)
         /// <summary>
-        /// The primary identifier of a key-value.
-        /// The key is used in unison with the label to uniquely identify a key-value.
+        /// The primary identifier of a configuration setting.
```
I've seen "Gets or sets " prefix being used for property documentation. We need to decide if we want to use it too.

jsquire replied:
One vote against from me.   It's  the default template for properties and seems to get left a lot, but I question what value it provides.   Personally, I view the default case as being a read/write state and comment when that is not the case.

pakrym commented on src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient.cs:
```diff
@@ -12,16 +12,28 @@
         private readonly HttpPipeline _pipeline;
 
+        /// <summary>
+        /// Creates a <see cref="ConfigurationClient"/> that sends requests to the configuration store.
```
`Initializes a new instance of the ` is often use as ctor doc prefix in corefx

maririos committed:
pr feedback

jsquire commented on src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/SettingSelector.cs:
```diff
@@ -24,7 +30,7 @@ public class SettingSelector
         public IList<string> Labels { get; set; }
         /// <summary>
-        /// IKeyValue fields that will be retrieved.
+        /// Specify which fields will be retrieved.
```
To me, the phrasing feels a bit awkward here.   Maybe we should consider something like `Controls the fields that will be populated` or `Allows requesting a specific set of fields` ?

jsquire commented on src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationSetting.cs:
```diff
@@ -9,13 +9,22 @@
 namespace Azure.ApplicationModel.Configuration
 {
+    /// <summary>
+    /// ConfigurationSetting is a resource identified by unique combination of key and label.
```
To me, the phrasing is a bit awkward.   Perhaps we should consider something like `A setting, defined by a unique combination of a key and label`.

jsquire commented on src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient.cs:
```diff
@@ -12,6 +12,9 @@
 namespace Azure.ApplicationModel.Configuration
 {
+    /// <summary>
+    /// ConfigurationClient can be used for all operations with a <see cref="ConfigurationSetting"/> in the Azure Configuration Store.
```
Maybe we should consider something like `The client to use for interacting with the Azure Configuration Store` or something similar?   The name of the class leading the sentence feels a bit awkward to me.

jsquire reviewed (approved):
I'm assuming that we're intending the comments as the basis for consumers to use with intellisense and, potentially, generate documentation from.

As a nit, we've got some inconsistent use of punctuation across these, where some sentences end with a period and others do not.   We may want to consider making those consistent.

We've also got some mixed approaches to summaries, where some are more sentences-lie and the others more phrase-like.  We may want to consider a unified voice so that they're consistent across.  Personally, I believe that they make better documentation when they read as true sentences and offer a bit of context as well as the basic description of "what this thing does."

maririos committed:
PR feedback

maririos commented:
> I'm assuming that we're intending the comments as the basis for consumers to use with intellisense and, potentially, generate documentation from.

Correct

> As a nit, we've got some inconsistent use of punctuation across these, where some sentences end with a period and others do not. We may want to consider making those consistent.

Thanks! Modified

> We've also got some mixed approaches to summaries, where some are more sentences-lie and the others more phrase-like. We may want to consider a unified voice so that they're consistent across. Personally, I believe that they make better documentation when they read as true sentences and offer a bit of context as well as the basic description of "what this thing does."

I see. Modified
